release.py

```diff
@@ -71,7 +160,10 @@ def chdir_to_repo_root():
 
         os.chdir(path)
 
-        def test_first_line(filename, test):
+        def test_first_line(
+            filename: str,
+            test: Callable[[str], re.Match[str] | None] | Callable[[object], bool],
```

Comment:
```suggestion
            test: Callable[[str], object],
```
Avoid overly specific type.